handlers/handlers.go

```diff
@@ -40,68 +42,98 @@ func setStatusCode(req *http.Request, w http.ResponseWriter, err error) {
 	w.WriteHeader(status)
 }
 
-func forwardFlorenceTokenIfRequired(req *http.Request) *http.Request {
-	if len(req.Header.Get(common.FlorenceHeaderKey)) > 0 {
-		ctx := common.SetFlorenceIdentity(req.Context(), req.Header.Get(common.FlorenceHeaderKey))
-		return req.WithContext(ctx)
-	}
-	return req
+func setStatusCode404(req *http.Request, w http.ResponseWriter, err error) {
+	status := http.StatusNotFound
+	log.ErrorCtx(req.Context(), err, log.Data{"setting-response-status": status})
+	w.WriteHeader(status)
 }
 
-//GeographyRender ...
-func GeographyRender(rend RenderClient) http.HandlerFunc {
+//GeographyHomepageRender ...
+func GeographyHomepageRender(rend RenderClient) http.HandlerFunc {
 
 	return func(w http.ResponseWriter, req *http.Request) {
 		var page geographyHomepage.Page
 
-		resp, err := http.Get(`https://api.dev.cmd.onsdigital.co.uk/v1/code-lists` + localAuthority)
+		homePageLink := url + `/code-lists` + localAuthority
+
+		resp, err := http.Get(homePageLink)
 		if err != nil {
+			log.Error(err, log.Data{"error getting data from the code-lists api http.Get(" + homePageLink + ") for GeographyHomepageRender returned ": err})
+			log.Error(err, log.Data{"test": "error http.Get"})
 			setStatusCode(req, w, err)
 			return
 		}
+		if resp.StatusCode == 404 {
+			log.Error(err, log.Data{"error getting data from the code-lists api http.Get(" + homePageLink + ") for GeographyHomepageRender returned ": resp.StatusCode})
+			setStatusCode404(req, w, err)
+			return
+		}
 		b, err := ioutil.ReadAll(resp.Body)
 		if err != nil {
+			log.Error(err, log.Data{"error getting the .Body data from the code-lists api http.Get(" + homePageLink + ") for GeographyHomepageRender .Body returned ": resp.Body})
 			setStatusCode(req, w, err)
 			return
 		}
 		var codelistresults models.CodeListResults
 		err = json.Unmarshal(b, &codelistresults)
 		if err != nil {
+			log.Error(err, log.Data{"error Unmarshaling the .Body data from the code-lists api http.Get(" + homePageLink + ") for GeographyHomepageRender returned ": err})
 			setStatusCode(req, w, err)
 			return
 		}
 		var codelist models.CodeList
 		err = json.Unmarshal(b, &codelist)
 		if err != nil {
+			log.Error(err, log.Data{"error Unmarshaling the .Body data from the code-lists api http.Get(" + homePageLink + ") for GeographyHomepageRender returned ": err})
 			setStatusCode(req, w, err)
 			return
 		}
 
-		geographyTypes := ""
+		var geographyTypes []geographyHomepage.Items
+		geographyTypesLabel := ""
+		geographyTypesID := ""
 		for i := range codelistresults.Items {
-			log.Debug("test", log.Data{
+			log.Debug("for loop", log.Data{
```

Comment:
If there is some usefulness in having a log for each iteration then could you update the log message?

Whenever I add a log message to my code I try to think in terms of a defect scenario i.e. if there was a bug here and another developer was trying to work out what was happening what info would help them get an idea of the state of the app.
If it's not needed just delete it.